kernel/sched.c

```diff
@@ -1564,6 +1566,9 @@ static int32_t z_tick_sleep(k_ticks_t ticks)
 	if (ticks > 0) {
 		return ticks;
 	}
+#else
+	/* busy wait to be time coherent since subsystems may depend on it */
+	z_impl_k_busy_wait(k_ticks_to_us_ceil32(expected_wakeup_ticks));
```

Comment:
huh, when did we start call those APIs in the code? You should be calling the actual API, not its implementation.